tvision/makes/common.imk

```diff
@@ -66,5 +66,5 @@ RHIDE_LIBS=$(addprefix -l,$(LIBS) $(RHIDE_OS_LIBS))
 # Same as RHIDE_LIBS, but without making reference to itself
 RHIDE_LIBS_ARCH=$(addprefix -l,$(RHIDE_OS_LIBS))
 RHIDE_LIBDIRS=$(LIB_DIRS)
-RHIDE_COMPILE_LINK=$(RHIDE_LD) $(RHIDE_LIBDIRS) $(LDFLAGS) $(C_EXTRA_FLAGS) -o $(OUTFILE)  $(OBJFILES) $(LIBRARIES) $(RHIDE_LIBS)
+RHIDE_COMPILE_LINK=$(RHIDE_LD) $(RHIDE_LIBDIRS) $(RHIDE_LDFLAGS) $(C_EXTRA_FLAGS) -o $(OUTFILE)  $(OBJFILES) $(LIBRARIES) $(RHIDE_LIBS)
```

Comment:
All other flags have an RHIDE_ prefix, so make LDFLAGS consistent.

Reply:
I'm not sure about it. Take a look at the definition:

```
RHIDE_COMPILE_ARCHIVE=$(RHIDE_LD) $(RHIDE_LIBDIRS) $(LDFLAGS) \
	$(RHIDE_LDFLAGS) $(C_EXTRA_FLAGS) -o $(OUTFILE)  \
	$(OBJFILES) $(RHIDE_LIBS_ARCH)
```

Here both are used: 
1. You can set `RHIDE_LDFLAGS`
2. You can add more in the *normal* way defining `LDFLAGS`

In your patch you are removing the user option to define extra stuff in `LDFLAGS`.

Why `RHIDE_COMPILE_LINK` doesn't contain `RHIDE_LDFLAGS`? This is commented as *Flags to link as a dynamic lib*

I think what you want is to add some new variable, something like `RHIDE_EXTRA_LDFLAGS` or similar. And then add it to the relevant definitions (RHIDE_COMPILE_ARCHIVE and RHIDE_COMPILE_LINK).
Adding RHIDE_LDFLAGS to RHIDE_COMPILE_LINK could introduce dynamic linking flags to the RHIDE_COMPILE_LINK sentence, which isn't right.

Reply:
I found the comment you refer to at line 256 on config.pl.  I think that comment simply means that the following section of code addresses libs for dynamic linking.  I don't think it any way implies that LDFLAGS or RHIDE_LDFLAGS address in particular dynamic linking or static linking.

We are stuck trying to make sense of what these variables are meant to do, without any clear documentation.  My interpretation after having worked with this code for just a short while is that the RHIDE variables like RHIDE_LDFLAGS are used so that the build files do not unintentionally overwrite commonly used counterparts like LDFLAGS, which would otherwise confuse the developer why the LDFLAGS he set are not being used.  That was my basis for switching from LDFLAGS to RHIDE_LDFLAGS.  I'm willing to admit I could be misinterpreting the original intent.

Reply:
One thing is the original idea, and another how it was used.
If you take a look at the config you'll see the comment really means what I told you: this variable is used for the dynamic link. This is because RHIDE never got to support the generation of dynamic libs. So I used what I had at hand.